From d3979f32904c2303db9e66eb0fc9d7d7cd6fc3d3 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:50 -0400
Subject: [usb] ehci: compute full-speed bandwidth usage correctly

Message-id: <1410887463-20674-39-git-send-email-dzickus@redhat.com>
Patchwork-id: 95107
O-Subject: [RHEL7 PATCH 038/591] USB: EHCI: compute full-speed bandwidth usage correctly
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 2b90f01b219e390e1f1bf68dd7a2333efb3e3eff
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Fri Oct 11 11:28:21 2013 -0400

    USB: EHCI: compute full-speed bandwidth usage correctly

    Although the bandwidth statistics maintained by ehci-hcd show up only
    in the /sys/kernel/debug/usb/devices file, they ought to be calculated
    correctly.  The calculation for full-speed isochronous endpoints is
    wrong; it mistakenly yields bytes per microframe instead of bytes per
    frame.  The "interval" value, which is in frames, should not be
    converted to microframes.

    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ehci-sched.c b/drivers/usb/host/ehci-sched.c
index 34b5945..4b0903c 100644
--- a/drivers/usb/host/ehci-sched.c
+++ b/drivers/usb/host/ehci-sched.c
@@ -1055,7 +1055,7 @@ iso_stream_init (
   } else
    stream->raw_mask = smask_out [hs_transfers - 1];
   bandwidth = stream->usecs + stream->c_usecs;
-  bandwidth /= interval << 3;
+  bandwidth /= interval;
 
   /* stream->splits gets created from raw_mask later */
   stream->address = cpu_to_hc32(ehci, addr);
-- 
1.7.1